lib/fabrik/DirectorInstance.js

```diff
@@ -50,32 +50,48 @@ class DirectorInstance extends BaseInstance {
         this.networkSegmentIndex = networkSegmentIndex;
       })
       .tap(() => {
+        //TODO: This still has coupling with space guid as it calls ensureSpaceGuid. 
+        //As part of SF 2.0, we can go to ETCD and fetch the tenant_guid which will 
+        //ensure we dont call CF again. For now, we add a check for the feature validation.
         if (operation.type === 'delete') {
           return Promise
             .all([
               this.deleteSecurityGroup(),
-              this.ensureSpaceGuid()
-              .then(space_guid => this.manager.deleteRestoreFile(space_guid, this.guid))
-              .catch(err => {
-                logger.error(`Failed to delete restore file of instance '${this.guid}'`);
-                logger.error(err);
-                throw err;
-              })
+              this.deleteRestoreFile()
```

Comment:
@jagadish-kb Could you please see if this looks good ? Here we only check of backup feature for deletion of restore file in a separate method which does the feature check.